packages/web/components/complex/paginated-table.tsx

```diff
@@ -0,0 +1,184 @@
+                  {header.isPlaceholder ? null : (
+                    <div
+                      {...{
+                        className: header.column.getCanSort()
+                          ? "cursor-pointer select-none"
+                          : "",
+                        onClick: header.column.getToggleSortingHandler(),
+                      }}
+                    >
+                      {flexRender(
+                        header.column.columnDef.header,
+                        header.getContext()
+                      )}
+                      {{
+                        asc: (
+                          <Image
+                            alt="ascending"
+                            src={
+                              IS_FRONTIER
+                                ? "/icons/sort-up-white.svg"
+                                : "/icons/sort-up.svg"
+                            }
+                            height={16}
+                            width={16}
+                          />
+                        ),
+                        desc: (
+                          <Image
+                            alt="descending"
+                            src={
+                              IS_FRONTIER
+                                ? "/icons/sort-down-white.svg"
+                                : "/icons/sort-down.svg"
+                            }
+                            height={16}
+                            width={16}
+                          />
+                        ),
+                      }[header.column.getIsSorted() as string] ?? null}
+                    </div>
+                  )}
+                </th>
+              );
+            })}
+          </tr>
+        ))}
+      </thead>
+      <tbody>
+        {paddingTop > 0 && (
+          <tr>
+            <td style={{ height: `${paddingTop}px` }} />
+          </tr>
+        )}
+        {virtualRows.map((virtualRow) => {
+          const row = rows[virtualRow.index] as Row<Pool>;
+          return (
+            <tr
+              key={row.id}
+              className="transition-colors focus-within:bg-osmoverse-700 focus-within:outline-none hover:cursor-pointer hover:bg-osmoverse-800"
+            >
+              {row.getVisibleCells().map((cell) => {
+                return (
+                  <td key={cell.id}>
+                    <Link
+                      href={`/pool/${row.original[0].poolId}`}
+                      key={virtualRow.index}
+                    >
+                      <a className="focus:outline-none">
+                        {flexRender(
+                          cell.column.columnDef.cell,
+                          cell.getContext()
+                        )}
+                      </a>
+                    </Link>
+                  </td>
+                );
+              })}
+            </tr>
+          );
+        })}
+        <div ref={loaderRef} />
```

Comment:
What is this used for? I got a warning that div cannot be used as child of tbody.

Reply:
This is used to trigger pagination, I use the `useOnScreen` hook to observe when the ref appears on screen, and run `fetchRemainingPools` when that happens since this is the last div in the table. I've fixed the console errors by moving this ref into the table row instead.